### ISSUE TITLE
Missing epoch_23_hvq_my_mvtecad.pth for Costfilter_HVQ_mvtec

### ISSUE THREAD

**Issue by A-Cycle-M** [ZHE-SAPI/CostFilter-AD#20]
I would like to test **Costfilter_HVQ_mvtec** by using "/CostFilter-AD/Costfilter_HVQtrans/Costfilter_HVQ_mvtec/experiments/MVTec-AD/eval.sh". However, I see that line 350 in "CostFilter-AD/Costfilter_HVQtrans/Costfilter_HVQ_mvtec/tools/train_val.py" uses:

`model_path = os.path.join('./checkpoint_paths/epoch_23_hvq_my_mvtecad.pth')`

But **epoch_23_hvq_my_mvtecad.pth** is missing

**Comment by ZHE-SAPI** [comment]
https://drive.google.com/drive/mobile/folders/1vp-Sp0LUFsRYHlz2KlQnQ03gh6DyZRO8?usp=sharing        this is the download link, pls refer to the readme file in https://github.com/ZHE-SAPI/CostFilter-AD/tree/main/Costfilter_HVQtrans， thanks

---- Replied Message ----
| From | ***@***.***> |
| Date | 11/13/2025 09:09 |
| To | ZHE-SAPI/CostFilter-AD ***@***.***> |
| Cc | Subscribed ***@***.***> |
| Subject | [ZHE-SAPI/CostFilter-AD] Missing epoch_23_hvq_my_mvtecad.pth for HVQ_mvtec (Issue #20) |
Amina-Mokhtar created an issue (ZHE-SAPI/CostFilter-AD#20)

I would like to test Costfilter_HVQ_mvtec by using "/CostFilter-AD/Costfilter_HVQtrans/Costfilter_HVQ_mvtec/experiments/MVTec-AD/eval.sh". However, I see that line 350 in "CostFilter-AD/Costfilter_HVQtrans/Costfilter_HVQ_mvtec/tools/train_val.py" uses:

model_path = os.path.join('./checkpoint_paths/epoch_23_hvq_my_mvtecad.pth')

But epoch_23_hvq_my_mvtecad.pth is missing

**Comment by Amina-Mokhtar** [comment]
Thank you for the fast reply!

Ah, I see, so it is the same as the one mentioned in the readme. I wasn't sure since the names are different. 

So I did run it using that pth file, but the results, especially the pred_gray, look a bit off. It might be just a plotting issue.

Good Example:

<img width="1534" height="230" alt="Image" src="https://github.com/user-attachments/assets/5c1abbce-0d32-4680-b161-f61798b3c81f" />


Broken:

<img width="1534" height="230" alt="Image" src="https://github.com/user-attachments/assets/c7479f81-a597-4e23-9354-66e5a9705621" />

**Comment by ZHE-SAPI** [comment]
yes, we can norm the pred grey by  (x-x.min)/(x.max-x.min) for better results

---- Replied Message ----
| From | ***@***.***> |
| Date | 11/13/2025 09:33 |
| To | ZHE-SAPI/CostFilter-AD ***@***.***> |
| Cc | Zhang ***@***.***>,
Comment ***@***.***> |
| Subject | Re: [ZHE-SAPI/CostFilter-AD] Missing epoch_23_hvq_my_mvtecad.pth for Costfilter_HVQ_mvtec (Issue #20) |
Amina-Mokhtar left a comment (ZHE-SAPI/CostFilter-AD#20)

Thank you for the fast reply!

Ah, I see, so it is the same as the one mentioned in the readme. I wasn't sure since the names are different.

So I did run it using that pth file, but the results, especially the pred_gray, look a bit off. It might be just a plotting issue.

Good Example:

Screenshot.from.2025-11-13.10-30-21.png (view on web)

Broken:

Screenshot.from.2025-11-13.10-30-30.png (view on web)

**Comment by ZHE-SAPI** [comment]
Thanks for the follow-up.

1) **Generalization / CIFAR results.**  
I’m not yet sure which dataset you trained the model on (let’s call it *Dataset A*) before testing on **CIFAR**. Conceptually, this setup is close to a *zero-shot anomaly detection* scenario (trained on Dataset A, tested on a different target distribution). In that case, a performance drop is expected, because the anomaly patterns and low-level statistics in CIFAR can be very different from those in Dataset A. In addition, there is a broad consensus in the community (e.g., **AdaCLIP**, **AnomalyCLIP**, **April-GAN**) that the choice of the auxiliary training dataset (Dataset A) can significantly affect the model’s generalization performance.


If your goal is stronger cross-dataset generalization, a more robust direction is to leverage **CLIP-based** frameworks (e.g., **AdaCLIP**, **AnomalyCLIP**, **April-GAN**) that rely on CLIP’s broad transfer ability. In fact, we have [integrated our **CostFilter** module into these CLIP-based baselines](https://arxiv.org/html/2510.03363v1) and observed better zero-shot performance than the original baselines. That said, strong generalization typically depends heavily on **CLIP-style zero-shot / test-time fine-tuning** (or lightweight adaptation), rather than purely training on a single dataset and directly transferring.

2) **MVTec “screw” visualization.**  
For the screw heatmap, I’d suggest first **normalizing the anomaly map** before visualization (otherwise the contrast can look misleading). Also, we have indeed seen some cases where the screw category is harder and the qualitative results are not always ideal—this class can be challenging due to subtle defect patterns and strong texture/geometry regularities.

Finally, it may also depend on the backbone: if you are using the **CostFilter-HVQTrans** variant, the visual quality can be limited by the underlying HVQTrans baseline. If possible, I recommend visualizing **CostFilter-Dinomaly** on the screw category as well; it often produces cleaner anomaly maps.
